### ISSUE TITLE
Running MT experiments with mlm_100_1280 (100 languages) model

### ISSUE THREAD

**Issue by Akella17** [facebookresearch/XLM#198]
@aconneau @glample Can one use the MLM model trained on 100 languages for multilingual unsupervised MT task? I am running into CUDA memory issues upon running this model. The model seems to be too large to load into a 2 GPU (12GB each) system. Is there any alternative for using this pretrained model?

Do you plan on releasing a small scale model trained for 100 languages in the future? Also, is it effective to use the intermediate layers (first 6 layers) of this large pretrained model to initialize the encoder? Also, would I need adaptive softmax, i.e., ```asm=True``` and ```share_inout_emb = False``` for training on a 200K vocab?

```
python train.py --exp_name unsupMT_enfr --dump_path ./dumped/ --reload_model 'mlm_100_1280.pth,mlm_100_1280.pth'
--data_path ./data/processed/en-fr/ --lgs 'en-fr' --ae_steps 'en,fr' --bt_steps 'en-fr-en,fr-en-fr'
--word_shuffle 3 --word_dropout 0.1 --word_blank 0.1 --lambda_ae '0:1,100000:0.1,300000:0'
--encoder_only false --emb_dim 1280 --n_layers 16 --n_heads 8 --dropout 0.1 --attention_dropout 0.1
--gelu_activation true --tokens_per_batch 100 --batch_size 32 --bptt 256
--optimizer adam_inverse_sqrt,beta1=0.9,beta2=0.98,lr=0.0001 --epoch_size 200000
--eval_bleu true --stopping_criterion 'valid_en-fr_mt_bleu,10'
--validation_metrics 'valid_en-fr_mt_bleu' --max_vocab 200000 --use_lang_emb False
```

**Comment by Lavine24** [comment]
@Akella17 Have you solve this problem? Recently, I also want to have some attempt in that scenario. I still do not known that how can we use the released pretrained model "mlm_100_1280.pth" to UNMT tasks, Can you give me some useful solutions for this cases. Thank you very much.

**Comment by Akella17** [comment]
@Lavine24 I did not find any solution to this problem. You can maybe use distillation (has been done for BERT and is straightforward to extend to XLM) to reduce the parameters of XLM100. However, I did not do it so I cannot guarantee anything.
